packages/material-ui/src/Button/Button.js

```diff
@@ -228,16 +240,15 @@ function Button(props) {
 
   const fab = variant === 'fab' || variant === 'extendedFab';
   const contained = variant === 'contained' || variant === 'raised';
-  const text = variant === 'text' || variant === 'flat' || variant === 'outlined';
```

Comment:
Removing `outlined` is ok but `flat` has to stay for now. It's the previous name of `text` and has to stay until the next major release.